src/backend/SentenceGenerator.ts

```diff
@@ -249,9 +249,25 @@ export class SentenceGenerator {
                 return ["", false];
             }
 
-            const mapping = this.ruleMappings?.[ruleName];
-            if (mapping) {
-                return [addSpace ? mapping + " " : mapping, false];
+            const mappingValue = this.ruleMappings?.[ruleName];
+            if (mappingValue) {
+                let mapping: string | undefined;
+                if (Array.isArray(mapping)) {
+                    if (mapping.length < 0) {
```

Comment:
Shouldn't that be `> 0`?

Reply:
crap, that is true!

Reply:
total lack of attention

Reply:
I guess part of the tests not passing was about that, and the other part due to the fact that I was returning empty string instead of doing a fall-back to the regular generation, which is done now.